### PR TITLE
Low level ed25519 arithmetics

### PR DESCRIPTION
**Pull request by lmctv**
Supersedes #408 adding tests and postponing crypto_core_ed25519_from_uniform.

### CODE REVIEW

File: tests/test_bindings.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
 from nacl.exceptions import BadSignatureError, CryptoError
 
 
+_BA = bytearray
```
Comment (reaperhulk) [inline tests/test_bindings.py]: I'd prefer not to alias like this

Comment (lmctv) [reply tests/test_bindings.py]: Removed the aliasing. This forced a reflow of the byte representation of the basepoint to please flake8.

**Comment by lmctv** [comment]
@reaperhulk, @pyca/pynacl-core: could someone please take another look at this PR?

**Comment by lmctv** [comment]
@reaperhulk do you think you could spare some time for another review of this PR?

**Comment by reaperhulk** [comment]
Just so you know I haven't forgotten this -- I will try to get to it soon. Sorry!

File: tests/test_bindings.py [hunk 2]
```diff
@@ -449,3 +449,69 @@ def test_sign_ed25519ph_libsodium():
...
+
+    p3 = c.crypto_core_ed25519_add(p1, p2)
+
+    assert c.crypto_core_ed25519_is_valid_point(p3) is True
+    assert c.crypto_core_ed25519_sub(p3, p1) == p2
+    assert c.crypto_core_ed25519_sub(p3, p2) == p1
+
+
+def test_scalarmult_ed25519():
+    SCALARBYTES = c.crypto_scalarmult_ed25519_SCALARBYTES
```
[the remark this answers is not in the document]
Comment (lmctv) [reply tests/test_bindings.py]: I hope the comments I just added are enough to clarify what happens in this test.

**Pull request merged**